doc/changes/8400.md

```diff
@@ -0,0 +1,2 @@
+- Add `--stop-on-first-error` option to `dune build` which will terminate the build when
+  the first error message is encountered. (#8400, @Alizter)
```

Comment:
Let's mention the name of whoever implemented the feature in the engine as well (pmwhite I believe)

Reply:
Yes, in fact I will probably remove my name completely if @pmwhite is happy with me writing that.

Reply:
You did some work too, so keep it.